pallets/loans-ref/src/types.rs

```diff
@@ -486,6 +486,21 @@ impl<T: Config> ActiveLoan<T> {
 		&self.write_off_status
 	}
 
+	/// Check if a write off rule is applicable for this loan
+	pub fn check_write_off_trigger(
+		&self,
+		trigger: &WriteOffTrigger,
+	) -> Result<bool, DispatchError> {
+		let now = T::Time::now().as_secs();
+		match trigger {
+			WriteOffTrigger::PrincipalOverdueDays(days) => {
+				let overdue_secs = SECONDS_PER_DAY.ensure_mul(days.ensure_into()?)?;
+				Ok(now >= self.maturity_date().ensure_add(overdue_secs)?)
+			}
+			WriteOffTrigger::OracleValuationOutdated(_seconds) => Ok(false),
+		}
+	}
```

Comment:
I've moved this to the `ActiveLoan` type because if it grows, it will need more info coming from the loan type (overdue days, oracle timestamp, interest payments, ...) so it makes sense to be a loan method.